app.py

```diff
@@ -1,13 +1,129 @@
 #!/usr/bin/python3
-from flask import Flask, render_template
+from flask import Flask, render_template, jsonify, request
+import pymysql
+from flask_cors import CORS, cross_origin
+import json
 import uuid
+from datetime import datetime
+import time
 
-application = Flask(__name__)
+app = Flask(__name__)
+cors = CORS(app, resources={r"/*": {"origins": "*"}})
 
-@application.route("/")
+@app.route('/', strict_slashes=False)
 def mainpage():
     return render_template('login.html',
-                           cache_id=uuid.uuid4())
+                           cache_id=uuid.uuid4()
+    )
 
-if __name__ == "__main__":
-    application.run(host='0.0.0.0')
+@app.route('/save_exp/', methods=['GET', 'OPTIONS', 'POST'], strict_slashes=False)
+def save_exp():
+    """ inserts new experience into 'experiences' table """
+    response = request.data.decode('utf-8')
+    obj = json.loads(response)
+
+    con = pymysql.connect('localhost', 'wellness_dev', 'wellness_dev_pwd', 'wellness_dev_db')
+    cursor = con.cursor()
+    exp_name = obj['exp_name']
+    scale = obj['scale']
+    if scale == '':
+        scale = 5
+    date = datetime.now().strftime('%Y-%m-%d 00:00:00')
+    _type = obj['type']
+    user_id = obj['user_id']
+    count = 0
+
+    cursor.execute('SELECT `count`, `date` FROM `experiences` WHERE exp_name=\'{}\' AND date=\'{}\' ORDER BY date DESC'.format(exp_name, date))
+
+    result = cursor.fetchall();
+    dates = [];
+
+    if not result:
+        count = 1
+    else:
+        for item in result:
+            if date == str(item[1]):
+                cursor.execute("UPDATE experiences set count=count+1 where date='{}'".format(date))
+                con.commit()
+                con.close()
+                return jsonify(True)
+            else:
+                dates.append(item[1])
+
+    cursor.execute('INSERT INTO experiences (exp_name, scale, date, type, user_id, count) VALUES("{}", "{}", "{}", "{}", "{}", "{}")'.format(exp_name, scale, date, _type, user_id, count))
+    con.commit()
+    con.close()
+
+    return jsonify(False)
+
+def data_exists(exp_name, date):
+    """ confirms existance of experience entry  in 'experiences table'"""
+    con = pymysql.connect('localhost', 'wellness_dev', 'wellness_dev_pwd', 'wellness_dev_db')
+    cursor = con.cursor()
+    results = ()
+    var1 = cursor.execute("SELECT EXISTS(SELECT 1 FROM experiences WHERE exp_name='{}' AND date='{}')".format(exp_name, date))
```

Comment:
Naomi built this then handed it off to me. I'm not sure we need it. If we do, I will update the variable names etc to be more consistent with our naming convention.  Otherwise I'll discard the method.